### DISCUSSION TITLE
Suggestion: Linter-like error highlighting

### DISCUSSION THREAD

**Discussion by ctcarton** [community/community#123770] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

There is a class of errors which neither the linter nor the editor can detect which would be very helpful if Copilot could. Here is one example where I've forgotten to save the parameter in my constructor (Typescript):

```ts
type MyParams = { ... }

class MyClass extends BaseClass {
  #params: MyParams

  constructor (params: MyParams & BaseParams) {
    super(params)
    // oops, forgot this.#params = params
  }
}
```

Neither the linter nor the editor would detect this problem, but Copilot could. It would be nice to have a warning if Copilot thinks that there is an error. It would probably need to be a very high confidence to avoid a lot of useless noise but if it worked properly it would be very useful.

**Comment by Akash1134** [comment]
Hi @ctcarton ,

We appreciate your feedback and your valuable input. We want to be transparent about our current priorities. You can gain insights into the projects we’re currently focusing on for the upcoming year by checking our [public roadmap](https://github.com/github/roadmap/projects/1). While smaller fixes may not always be explicitly listed, they often get addressed when related work is in progress. Also, with the rapidly changing AI 

Regarding your specific request, it’s important to note that it’s not actively being worked on at the moment. However, this doesn’t mean we won’t consider it in the future; it’s simply not on our development roadmap right now.

We’d like to emphasize that as a valued GitHub user, your feedback in this Discussion space, along with appropriate tags and examples, plays a crucial role in shaping the development of GitHub. While we can’t guarantee immediate action on every piece of feedback, please rest assured that we are attentive to your suggestions and continuously strive to enhance the GitHub experience.
While we do review all incoming feedback unfortunately due to the volume of feedback, we may not be able to respond to each one individually at this time. However, we strongly encourage you to continue providing your valuable input as it greatly helps us prioritize and improve GitHub.

Your contributions are instrumental in making GitHub better for all users. Thank you for taking the time to share your thoughts with us, even if the particular request is not currently in active development. Your input is truly valuable and contributes to the continuous improvement of GitHub.
